src/integrationtests/java/com/aws/greengrass/integrationtests/ShadowManagerTest.java

```diff
@@ -247,4 +264,63 @@ void GIVEN_shadow_manager_WHEN_strategy_config_resets_THEN_respond_to_config_upd
         kernel.getContext().waitForPublishQueueToClear();
         assertThat(syncHandler.getOverallSyncStrategy(), instanceOf(PeriodicSyncStrategy.class));
     }
+
+    @Test
+    void GIVEN_shadow_manager_running_AND_crypto_service_loading_WHEN_sync_requests_with_dataplane_client_THEN_retry_requests_with_good_client_after_crypto_service_loads(ExtensionContext context)
+            throws InterruptedException, TLSAuthException, IoTDataPlaneClientCreationException {
+        ignoreExceptionOfType(context, TLSAuthException.class);
+        ignoreExceptionOfType(context, RetryableException.class);
+
+        SecurityService securityService = mock(SecurityService.class);
+        when(securityService.getDeviceIdentityKeyManagers()).thenThrow(TLSAuthException.class);
+        kernel.getContext().put(SecurityService.class, securityService);
+
+        IotDataPlaneClientFactory factory = kernel.getContext().get(IotDataPlaneClientFactory.class);
+        IotDataPlaneClientWrapper wrapper = spy(new FakeIotDataPlaneClientWrapper(factory));
```

Comment:
Instead of the full blown class below since we just overriding a single method we could

```java
spy(new IotDataPlaneClientWrapper(factory) {
      @Override
      public GetThingShadowResponse getThingShadow(String thingName, String shadowName) throws IoTDataPlaneClientCreationException {
         iotDataPlaneClientFactory.getIotDataPlaneClient(); // throws an IoTDataPlaneClientCreationException given security service is not ready
         return GetThingShadowResponse.builder().build();
     }
});
```